### PR TITLE
Move wiki documents to new style/ directory

### PR DESCRIPTION
**Pull request by nomicode**
the wiki pages should be deleted when this PR is merged

### CODE REVIEW

File: style/rst.rst [hunk 1]
```diff
@@ -0,0 +1,464 @@
+==============
+RST Guidelines
+==============
+
+At Crate.io, our documentation is written in RST. In order to create a uniform
+standard for writing our RST, we have developed some guidelines. Following
+these guidelines is required for contributions to our documentation. For ease
+of use, we have gathered them in this wiki page below.
+
+**Note**:
```
Comment (matthijskrul) [inline style/rst.rst]: This should no longer be here! This was only relevant for Github wiki

File: style/rst.rst [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+General
+=======
+
+With the exception of pre-formatted text (e.g., literals and code blocks), as
+well as link URLs, lines should not exceed 79 characters. They should be
+wrapped as closely to 79 characters per line as possible, without exceeding
+that limit.
+
+**Tip**:
```
Comment (matthijskrul) [inline style/rst.rst]: We can also now use Sphinx theme things for tips, warnings, etc again (I just used bold for the wiki)

**Review by matthijskrul — CHANGES_REQUESTED** [review]

**Comment by nomicode** [comment]
@matthijskrul addressed your comments!

looks like GitHub RST viewing doesn't properly support admonition styles, which sucks. but I'm happy with this as an interim solution

**Comment by matthijskrul** [comment]
> @matthijskrul addressed your comments!
> 
> looks like GitHub RST viewing doesn't properly support admonition styles, which sucks. but I'm happy with this as an interim solution

Yeah it has various limitations, also with certain links and such. So we have to choose between adjusting for that (with bolding or somesuch) or using them anyway with the premise that people will get it and so we could easily run it through Sphinx later on without extra effort. I suspect the latter is the more robust one but it's up to you! I just wanted to flag it.

**Review by matthijskrul — APPROVED** [review]

**Pull request merged**